server/src/main/java/org/opensearch/tracer/SpanHolder.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracer;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+public class SpanHolder {
+
+    private final AtomicReference<Span> span = new AtomicReference<>();
+
+    public SpanHolder(SpanHolder spanHolder) {
+        this.span.getAndSet(spanHolder.span.get());
```

Comment:
Why do we need getAndSet here? only set should work. all the places in this class.

Reply:
done